sympy/calculus/tests/test_util.py

```diff
@@ -179,6 +179,25 @@ def test_periodicity_check():
     assert periodicity(Abs(sec(sec(x))), x) == pi
 
 
+def test_periodicity_Function():
+    x = Symbol('x')
+    class f(Function):
+        nargs = 1
+        def period(self, sym):
+            arg = self.args[0]
+            return 1/arg.coeff(sym)
+
+    assert periodicity(f(x), x) == 1
+    assert periodicity(f(2*x), x) == S(1)/2
+    assert periodicity(2*f(x), x) == 1
+    assert periodicity(x*f(2*x), x) is None
+    assert periodicity(f(x)*f(x), x) == 1
+    assert periodicity(exp(f(x)), x) == 1
+    assert periodicity(sin(x*pi)*f(x), x) == 2
+
+
+
+
```

Comment:
I don't think that there should be so many blank lines here

Reply:
Sorry. Fixed this.